test/sni_utility_test.cc

```diff
@@ -0,0 +1,47 @@
+#include "external/envoy/test/test_common/utility.h"
+
+#include "common/uri_impl.h"
+
+#include "client/sni_utility.h"
+
+#include "gtest/gtest.h"
+
+using namespace testing;
+
+namespace Nighthawk {
+namespace Client {
+
+class SniUtilityTest : public Test {
+public:
+  void checkSniHostComputation(const std::vector<std::string>& uris,
+                               const std::vector<std::string>& request_headers,
+                               const Envoy::Http::Protocol protocol,
+                               absl::string_view expected_sni_host) {
+    std::vector<UriPtr> parsed_uris;
+    parsed_uris.reserve(uris.size());
+    for (const std::string& uri : uris) {
+      parsed_uris.push_back(std::make_unique<UriImpl>(uri));
+    }
+    std::string sni_host = SniUtility::computeSniHost(parsed_uris, request_headers, protocol);
+    EXPECT_EQ(sni_host, expected_sni_host);
```

Comment:
Calling these macros from helper functions tends to complicate test failure readability. I seem to remember that the test failures will point to this line and might not show a good view of the expected and actual values. Could you verify that by making one of these fail?

Would it be better to update this helper function to just construct the parsed_uri? We could then call both the function under test and the comparison macro (EXPECT_EQ) from the test body. That would also result in a more typical test.

Reply:
amded in d915684fbbfd81630369b6fe99444df737a4ddc6